PKPD/gui/simulation.py

```diff
@@ -24,70 +26,126 @@ def __init__(self, main_window):
 
         # plot data
-        if self.is_single_output_model: # single output
+        if self.is_single_output_model:  # single output
             # clear figure
             self.data_model_figure.clf()
 
             # create plot
             self.data_model_ax = self.data_model_figure.subplots()
-            self.data_model_ax.scatter(x=self.time_data, y=self.state_data, label='data', marker='o', color='darkgreen',
-                                       edgecolor='black', alpha=0.5)
+
+            # iterate over each patient and add their data to the scatter plot
+            for patient_id in self.patients_data:
+                label = 'patient ' + str(patient_id)
+                colour = int(len(self.colours) * (patient_id - self.first_patient_id) / len(self.patients_data))
+                self.data_model_ax.scatter(x=self.patients_data[patient_id][0],
+                                           y=self.patients_data[patient_id][1][:, 0], label=label, marker='o',
+                                           color=self.colours[colour],
+                                           edgecolor='black', alpha=0.5)
             self.data_model_ax.set_xlabel(time_label)
-            self.data_model_ax.set_ylabel(state_label)
+            self.data_model_ax.set_ylabel(state_labels[0])
+            self.data_model_ax.set_xlim(left=0.0)
+            self.data_model_ax.set_ylim(bottom=0.0)
             self.data_model_ax.legend()
 
             # refresh canvas
             self.canvas.draw()
-        else: # multi output
+        else:  # multi output
             # clear figure
             self.data_model_figure.clf()
 
             # create subplots for each compartment
             self.data_model_ax = self.data_model_figure.subplots(nrows=self.data_dimension, sharex=True)
+
             for dim in range(self.data_dimension):
-                self.data_model_ax[dim].scatter(x=self.time_data, y=self.state_data[:, dim], label='data', marker='o',
-                                                color='darkgreen', edgecolor='black', alpha=0.5)
+
+                # iterate over each patient and add their data to the scatter plot
+                for patient_id in self.patients_data:
+                    print(patient_id)
+                    label = 'patient ' + str(patient_id)
+                    colour = int(len(self.colours) * (patient_id - self.first_patient_id) / len(self.patients_data))
+                    self.data_model_ax[dim].scatter(x=self.patients_data[patient_id][0],
+                                                    y=self.patients_data[patient_id][1][:, dim], label=label,
+                                                    marker='o',
+                                                    color=self.colours[colour],
+                                                    edgecolor='black', alpha=0.5)
                 self.data_model_ax[dim].set_ylabel(state_labels[dim])
                 self.data_model_ax[dim].legend()
```

Comment:
If you set this to just plot the legend for the first subplot, it should get round the issues with legends all over the place. You can also set the legend to be off on one side with the loc argument (sets where in fig) and bbox arguments (sets exact location of top left of legend), e.g.:
`legend(loc = 'upper right', bbox_to_anchor=(0.5, 0.5))`